src/main/scala/org/deeplearning4j/scalphagozero/board/GameState.scala

```diff
@@ -45,11 +53,7 @@ class GameState(
       move match {
         case Move.Play(point) =>
           val nextBoard = this.board.clone()
-          try {
-            nextBoard.placeStone(nextPlayer, point)
-          } catch {
-            case _: Exception => println(" Illegal move attempted at: " + point.toCoords)
```

Comment:
I put this in as safeguard, as sometimes in longer games this happened. it's obviously from an underlying bug.

Are you certain we can remove this?

Reply:
>  Are you certain we can remove this?

No, I'm not. It's the problem with exceptions. We can never be 100% confident when removing them because their are not typesafe and they break referential transparency. We tend not to use them in FP.

How are you familiar with FP and Scala ? 
I don't want to be cocky to you. You're the smart man here. I'm just someone rewriting code. A good AI could do what I'm doing.

I removed it because the obvious use of this guard was to catch an exception you thrown in the `placeStone` method, which I just replace by the goal of this `throw IllegalStateEx...`: print a message. (It also breaks referential transparency but I can improve that later)

Can we remove this guard and try to reproduce the mentioned to find its root and fix it ?

>  it's obviously from an underlying bug.

Do you have an idea where this bug can come from ?

Reply:
yeah, I'm aware of all this. Sometimes breaking a paradigm can be a good thing, too. Note that with that exception the game flow continues and can still produce useful games, whatever the bug.

Not sure what's going wrong, I've tested a good chunk of GoBoard functionality, but a good candidate is what you just found: if previousStates are tracked incorrectly then it's no wonder there are attempts at playing moves that are in fact already on the board. In ZeroAgent I even filter by valid moves, but that doesn't help if the move tracker has an issue.